### DISCUSSION TITLE
How I create and use a 3x3-LED object?

### DISCUSSION THREAD

**Discussion by Lebostein** [pybricks/support#525] [Q&A] [answered]
Hi,

I saw an example with the new Spike 3x3-light matrix on the pybricks channel on YouTube. How I use this light matrix with pybricks? How I can find a code example?

Thanks!

**Comment by dlech** [comment]
It doesn't appear that we have added to the documentation yet, but there is a `LightMatrix` class in the `pupdevices`. You will need to use https://beta.pybricks.com to use this.

**Comment by laurensvalk** [comment, marked as answer]
You can use this example, adapted from #440.
```python
from pybricks.pupdevices import ColorLightMatrix
from pybricks.parameters import Port, Color
from pybricks.tools import wait

from urandom import randint

color_matrix = ColorLightMatrix(Port.B)

# Static single color example
color_matrix.on(Color.ORANGE)
wait(1000)

# Static multi color example
color_matrix.on([Color.RED, Color.WHITE, Color.BLUE] * 3)
wait(1000)

# Fade example
for i in range(3):
    for brightness in tuple(range(0, 110, 10)) + tuple(range(100, -10, -10)):
        color_matrix.on(Color.BLUE * (brightness / 100) )
        wait(50)

# Randomness
for i in range(1000):
    color_matrix.on([Color(randint(0, 360), v=40) for i in range(9)])
    wait(200)
```

**Reply by Lebostein** [reply]
Thanks!

Is there a documentation how I can install the pybricks firmware on the Spike Hub?
If you click on "Installing Pybricks on the SPIKE Prime Hub and the MINDSTORMS Inventor Hub" on the web page you end up in a dead end ("Please check back later"): http://pybricks.com/install/spike-mindstorms/
